rhessys/util/readtag_worldfile.c

```diff
@@ -89,8 +89,9 @@ param *readtag_worldfile(int *paramCnt, FILE *file,char *key){
 	    if(strcmp(strbuf2,"n_basestations")==0){
 	      break;
 	    }
+	    
```

Comment:
readtag_worldfile should be refactored to store variables for a given level in a hash table instead of an array.  As it stands now, for each variable in a level, we are doing a linear search of the array of tags, which has a worst-case complexity of N^2.  For basins, zones and hillslopes this isn't so bad, but for patches and strata, this is terrible as there are hundreds of variables at these levels, and potentially hundreds of thousands of patches.  The current solution will perform poorly for high resolution/large spatial extent worldfiles (increasing model start up time).  

I implemented a very simplistic hash table for [createflowpaths](https://github.com/RHESSys/RHESSys/blob/master/cf/src/patch_hash_table.c), which dramatically improved performance (~100x) when searching for patches.